### DISCUSSION TITLE
Can I control file existence based on OS?

### DISCUSSION THREAD

**Discussion by solzard** [twpayne/chezmoi#1708] [Q&A] [answered]
Please forgive me if I overlooked the documents. 

I have some app-config files that run only on macOS. Can I make them deploy on macOS but not on Linux? 
I don't want these files to appear on the home directories of remote Linux machines. Is this possible with templates or other means?

```shell
# Expected behavior
# On macOS
~/.mac-specific-app.cfg
~/.local/share/chezmoi/dot_mac-specific-app.cfg

# On Linux
~/.local/share/chezmoi/dot_mac-specific-app.cfg

**Comment by twpayne** [comment, marked as answer]
See [how to ignore files or a directory on different machines](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#ignore-files-or-a-directory-on-different-machines). Here's my [personal `.chezmoiignore`](https://github.com/twpayne/dotfiles/blob/master/home/.chezmoiignore).

**Reply by solzard** [reply]
Thank you, @twpayne!

**Reply by notDavid** [reply]
> See [how to ignore files or a directory on different machines](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#ignore-files-or-a-directory-on-different-machines).

For other people stumbling upon this, here's the current working link to the docs: [Use completely different dotfiles on different machines](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/#use-completely-different-dotfiles-on-different-machines)
